jax_galsim/interpolant.py

```diff
@@ -1464,8 +1464,11 @@ def tree_flatten(self):
     @classmethod
     def tree_unflatten(cls, aux_data, children):
         """Recreates an instance of the class from flattened representation"""
-        n = aux_data.pop("n")
```

Comment:
This line made changes to aux_data and that caused failures. The new method does not do this.